velox/common/memory/MallocAllocator.cpp

```diff
@@ -221,6 +221,19 @@ int64_t MallocAllocator::freeNonContiguous(Allocation& allocation) {
     Allocation::PageRun run = allocation.runAt(i);
     numFreed += run.numPages();
     void* ptr = run.data();
+    // Check if ptr was split into multiple PageRuns.
+    while (i + 1 < allocation.numRuns()) {
+      Allocation::PageRun nextRun = allocation.runAt(i + 1);
+      void* nextPtr = nextRun.data();
+      // NOTE: std::malloc will not return two allocated buffers which are
```

Comment:
Looks like this assumption is wrong. This is not the case with jemalloc looks like. CC: @kevinwilfong